git-hooks/post-checkout

```diff
@@ -1,3 +1,2 @@
-#!/usr/bin/env sh
-
-tar -C ./Carthage -xzf ./Carthage/Build.tar.gz
+#!/usr/bin/env bash
+exec git submodule update
```

Comment:
`submodule sync` is required if the `.gitmodules` file has changed - it defines the mapping of a submodule repo to a folder. I wouldn't include it. If that changes we'll need to manually coordinate it with the devs.